### ISSUE TITLE
[CLI]: wandb resume can not log if `step` is specified

### ISSUE THREAD

**Issue by qsh-zh** [wandb/wandb#5632]
### Describe the bug

```python
import wandb
run = wandb.init(project='resume_runs')
id = run.id
for i in range(5):
    run.log({'metric_A':i}, step=i)
run.finish()

# ! the following `metric_b` will not be logged
run_1 = wandb.init(project='resume_runs', id=id, resume='must')
for i in range(5):
    run_1.log({'metric_b':5+i}, step=i)
run_1.finish()
```

```shell

```

### Environment

WandB version: 0.14.1 and 0.15.3

OS: ubuntu 20.04

Python version: python3.8 / python3.9

Versions of relevant libraries: No

**Comment by qsh-zh** [comment]
The pattern is quite common in training and evaluating ML-models, where 'metric_A' may be for training loss and 'metric_b' the some evaluation metric after training and the 'step' is iteration number for the trained model.

**Comment by thanos-wandb** [comment]
Hi @qsh-zh thanks for reporting this issue, and the additional context. What's currently allowed is to log the resumed run metrics in a monotonically increasing time step, such as:

```
run_1 = wandb.init(project='resume_runs', id=id, resume='must')
for i in range(5):
    run_1.log({'metric_b':5+i}, step=5+i)
run_1.finish()
```

or, to only update summary metrics (instead of history), as follows:

```
run_1 = wandb.init(project='resume_runs', id=id, resume='must')
run_1.log({'metric_b': value})
run_1.finish()
```

May I please ask if it would help here to log the training and evaluations metrics in two different runs?

**Comment by qsh-zh** [comment]
@thanos-wandb , I want to make sure the evaluation metric is aligned with the training step~(saved checkpoints). It does not make sense that I upload evaluation results with step $5+i$  while the training iteration of the checkpoint is actually $i$.

Notice we do not overwrite an existing metric, e.g. 'metric_A', but add a metric that does not exist before.

**Comment by thanos-wandb** [comment]
Thank you @qsh-zh I agree that would be really useful functionality. I have increased the requests for this feature about updating the runs history store. I will temporarily close this ticket for now, and we will reach out to you here once there's any update on this.
